package/src/MessageBase.cpp

```diff
@@ -10,9 +63,12 @@ void MessageBase::log_init(const std::string& classname,
   // Use the sink list to create multi sink logger
   logger =
       std::make_shared<spdlog::logger>(classname, s_list.begin(), s_list.end());
-  log_level = level;
-  set_log_level(level);
-  logger->flush_on(level);
+  if (monitor_level.find(classname) != monitor_level.end()) {
+    log_level = monitor_level[classname];
+  } else
+    log_level = level;
+  set_log_level(log_level);
```

Comment:
The code indentation looks a bit broken here - remember to run the `clang_format` target